common/internal_model/src/main/scala/uk/ac/wellcome/models/pipeline/SourcePayload.scala

```diff
@@ -0,0 +1,37 @@
+package uk.ac.wellcome.models.pipeline
+
+import uk.ac.wellcome.storage.s3.S3ObjectLocation
+
+sealed trait SourcePayload {
+  val id: String
+  val version: Int
+}
```

Comment:
nit: I am a bit unsure why this is defined as a common interface / ADT rather than parameterically:

```scala
case class SourcePayload[T] {
  id: String,
  version: Int,
  payload: T
```

Both ways will work of couse, but the way it is defined currently indicates to me scenarios where we receive a generic `SourcePayload` and do something on it at runtime (like pattern match on it, or pull out the `id` or `version` defined on the base interface), where our transformers always know at compile time what sort of messages it receives.

Reply:
This makes sense to me, although looking at the classes it looks like the Miro data wouldn't fit this model and would need a migration. Probably worth doing anyway and use just a single case class with type parameter

Reply:
The problem is that the DynamoDB rows don’t all fit that pattern. My plan is to validate, but pass along the JSON unmodified. The reindexer output won’t change at all, but we’ll know it can be deserialised as the appropriate type.

I should ditch the sealed trait; it’s not adding anything.

Reply:
Oh you mean it doesn't fit if we want this to exactly match the columns defined in Dynamo? It would fit if we transformed each row first, but that would require the reindexer to know about these types (which I don't know if that is the plan?)

Reply:
I think it’ll make sense when I put it alongside the reindexer patch, but I didn’t finish it in time.

Reply:
Yep that makes more sense to me now :+1: . Sealed trait might still be useful if we want the `TransformerWorker` to receive some `Payload <: SourcePayload`.

Reply:
It could fit if we transformed each row first, yeah; I’m not sure if that adds much value IMO. You’d add a bunch of extra code to rename a couple of fields.